Wdivision-by-zero in s_security.cpp

clang version 22.1.2
Target: x86_64-w64-windows-gnu

```
/home/stefan/mt-build/build/64/luanti/src/script/cpp_api/s_security.cpp:1156:12: warning: remainder by
      zero is undefined [-Wdivision-by-zero]
 1156 |         t = t - t % (SSCSM_CLOCK_RESOLUTION_US * CLOCKS_PER_SEC / 1'000'000);
      |                   ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/script/cpp_api/s_security.cpp
+++ b/src/script/cpp_api/s_security.cpp
@@ -1153,7 +1153,9 @@ int ScriptApiSecurity::sl_os_setlocale(lua_State *L)
 int ScriptApiSecurity::sl_os_clock(lua_State *L)
 {
 	auto t = clock();
-	t = t - t % (SSCSM_CLOCK_RESOLUTION_US * CLOCKS_PER_SEC / 1'000'000);
+	const clock_t clock_resolution =
+		(SSCSM_CLOCK_RESOLUTION_US * CLOCKS_PER_SEC + 1'000'000 - 1) / 1'000'000;
+	t = t - t % clock_resolution;
 	lua_pushnumber(L, static_cast<lua_Number>(t) / static_cast<lua_Number>(CLOCKS_PER_SEC));
 	return 1;
 }

--- a/src/script/cpp_api/s_security.cpp
+++ b/src/script/cpp_api/s_security.cpp
@@ -1153,7 +1153,7 @@ int ScriptApiSecurity::sl_os_setlocale(lua_State *L)
 int ScriptApiSecurity::sl_os_clock(lua_State *L)
 {
 	auto t = clock();
-	const clock_t clock_resolution =
+	constexpr clock_t clock_resolution =
 		(SSCSM_CLOCK_RESOLUTION_US * CLOCKS_PER_SEC + 1'000'000 - 1) / 1'000'000;
 	t = t - t % clock_resolution;
 	lua_pushnumber(L, static_cast<lua_Number>(t) / static_cast<lua_Number>(CLOCKS_PER_SEC));

